### ISSUE TITLE
[Navigation Drawer] Not only the selected item shows bold when scrolling

### ISSUE THREAD

**Issue by T1mmyK0** [material-components/material-components-android#3191]
**Description:** Since version 1.8.0 the active navigation drawer item name shows bold. However, there's a bug that when scrolling, sometimes the name of random items get bold too. Usually the one from which you start scrolling (but not always. Sometimes two random items get bold) and only if you scroll it out of the screen. You need to have more items than fit on the screen to reproduce the bug.

**Expected behavior:** When scrolling only the active navigation drawer item should have a bold name. All other items should appear with normal text style.

**Source code:** I don't know, but in 1.7.0 the issue didnt exist yet, because there was no bold active item name at all.

**Android API version:** Android API 33

**Material Library version:** 1.8.0

**Device:** Samsung Galaxy S22 Ultra

**Comment by VicMikhailau** [comment]
The same issue.

**Comment by drchen** [comment]
We probably need to reset the typeface in `setTextAppearanceInactive()`.

**Comment by paulfthomas** [comment]
Hi @T1mmyK0, can you provide a screenshot/video and code sample for us to reproduce the issue? Thank you!

**Comment by T1mmyK0** [comment]
https://user-images.githubusercontent.com/58298629/217290662-6c05097c-4a88-4a30-9a3a-c8dbe3bb9728.mp4

@paulfthomas You see the effect after 10 seconds.

**Comment by paulfthomas** [comment]
@T1mmyK0 could you share some code samples? Thanks!

**Comment by T1mmyK0** [comment]
@paulfthomas What code samples? You only need to use the Android Studio navigation drawer creator and add like around 10 dummy menu items and set the first as the active one. Then you have a menu like mine. I can't share source code of my app.

**Comment by paulfthomas** [comment]
@T1mmyK0 we're having a hard time reproducing the exact issue, that's why I was asking for a code sample. Would it be possible for you to create a repo with a small sample app that shows the issue so it'd be easier for us understanding what's going on?

**Comment by VicMikhailau** [comment]
@paulfthomas 
It was really hard to reproduce that behavior on the default sample. I think main "rule breaker" here is the items grouping into the sections (group 1, group 2 and etc.). There is the sample screenshot and sample app (in app using Theme.MaterialComponents.Light.NoActionBar.Bridge).

[MyApplication.zip](https://github.com/material-components/material-components-android/files/10735799/MyApplication.zip)

![1](https://user-images.githubusercontent.com/19249482/218826354-88e77b4c-a83a-472e-9666-0aa7bcca1cd4.jpg)

**Comment by T1mmyK0** [comment]
@VicMikhailau Thank you for your help :)

**Comment by paulfthomas** [comment]
@VicMikhailau thanks so much, we'll try to get a fix soon!

**Comment by paulfthomas** [comment]
@T1mmyK0 @VicMikhailau are you both using `Theme.MaterialComponents.Light.NoActionBar.Bridge`? It seems related to this Bridge theme which is not actively supported anymore.

**Comment by VicMikhailau** [comment]
@paulfthomas in My case - yes.

**Comment by T1mmyK0** [comment]
@paulfthomas I have actually never heard of the Bridge theme. I have searched my whole project using the find in files option, but it hasn't found anything. So it's probably not related to the Bridge theme as I'm not using it at all.

**Comment by VicMikhailau** [comment]
@paulfthomas 
I have checked default themes for the app and changed `Bridge` theme to the usual `MaterialComponent/Material3`. It seems like you can reproduce that similar behavior through appling to the main theme some default font family like avenir_medium (`<item name="android:fontFamily">@font/avenir_medium</item>`).
Please apply/change in my sample (`themes.xml`) following code and you can see strange items blinking and described issue:

 ```
   <style name="Theme.MyApplication" parent="Theme.Material3.Light.NoActionBar">
        <!-- Primary brand color. -->
        <item name="colorPrimary">@color/purple_500</item>
        <item name="colorPrimaryVariant">@color/purple_700</item>
        <item name="colorOnPrimary">@color/white</item>
        <!-- Secondary brand color. -->
        <item name="colorSecondary">@color/teal_200</item>
        <item name="colorSecondaryVariant">@color/teal_700</item>
        <item name="colorOnSecondary">@color/black</item>
        
        <item name="android:fontFamily">@font/avenir_medium</item> <!-- Rule braker -->
        
        <!-- Status bar color. -->
        <item name="android:statusBarColor">?attr/colorPrimaryVariant</item>
        <!-- Customize your theme here. -->
    </style>
```

**Comment by paulfthomas** [comment]
Can you try to set `app:itemTextAppearance="@null"` on the `NavigationView` to see if it helps?

**Comment by T1mmyK0** [comment]
@paulfthomas For me it didn't help.

**Comment by VicMikhailau** [comment]
@paulfthomas Also still reproduced for the Main project and Sample.
Interesting - for the Sample project appling `app:itemTextAppearance="@null"` lead to the better visual issue representation.

**Comment by TWiStErRob** [comment]
I've also observed this in some way, and I also have two groups:
```xml
<menu>
	<group android:checkableBehavior="single">
		<item />
		<item />
		<item />
		<item />
	</group>
	<group android:checkableBehavior="single">
		<item />
		<item />
		<item />
		<item />
	</group>
</menu>
```
for me it wasn't when scrolling though, but when I leave the activity and then come back to it.
There's something checked in the first group and that shows active. Then in the second group I select something which goes to another activity. When I come back, the item corresponding to the item clicked in the second group is highlighted in the first group.

When I use `nav.setCheckedItem(id)`, everything is ok, because it de-activates the previously highlighted item.

Also, commit that added the bold: https://github.com/material-components/material-components-android/commit/b1bd580d7e6a587f88da5c034107bed1e77cd26a
it's probably too hacky, not customizable (textappearance attribute / state list etc?) and doesn't play well with recycling and state restoration.

@drchen Where's this bold requirement coming from btw, I don't see it in m2/m3 specs?

**Comment by yccheok** [comment]
Using

    implementation 'androidx.appcompat:appcompat:1.6.1'
    implementation 'com.google.android.material:material:1.8.0'

Observing the same bug (Text "Notes" remains bold even it is not currently being selected)

![Screenshot_1681536941](https://user-images.githubusercontent.com/308276/232185380-d20e4af4-1ada-40b7-b9b1-52ff4bce8c54.png)

    <menu xmlns:android="http://schemas.android.com/apk/res/android"
        xmlns:tools="http://schemas.android.com/tools"
        tools:showIn="navigation_view">

        <group android:checkableBehavior="single"
            android:id="@+id/top_group">
            <item
                android:id="@+id/nav_notes"
                android:icon="@drawable/ic_content_paste_black_24dp"
                android:title="@string/nav_notes" />
        </group>

        <group android:checkableBehavior="single"
            android:id="@+id/middle_group">
            <item
                android:id="@+id/nav_archive"
                android:icon="@drawable/ic_archive_black_24dp"
                android:title="@string/nav_archive" />
            <item
                android:id="@+id/nav_trash"
                android:icon="@drawable/ic_delete_black_24dp"
                android:title="@string/nav_trash" />
        </group>

Using multiple groups is important so that we can have a separator.

**Comment by davvelar** [comment]
@paulfthomas 
Very easy for me to reproduce using a modified version of the sample project. I just added a few more destinations to the menu, no scrolling needed. When picking destinations in the drawer others are becoming bold or normal according to a strange pattern. E.g If I continuously pick the top destination it removes the boldness of the titles of other destinations from the bottom up. If I instead continuously pick the bottom destination, it adds boldness of the titles of other destinations from the bottom up.

Using 
`implementation 'com.google.android.material:material:1.9.0'`
Run on a Pixel 3.

[MyApplication2.zip](https://github.com/material-components/material-components-android/files/11439271/MyApplication2.zip)

![Screenshot_20230510-092700](https://github.com/material-components/material-components-android/assets/96117990/898d4986-5c54-4145-820f-891e51d8aabc)

**Comment by MatthewCallery** [comment]
Downgrading to `com.google.android.material:material:1.7.0` is a temporary fix for this issue.

**Comment by stefanobreja** [comment]
@paulfthomas this is still happening when upgrading from 1.7.0 to any newer version

**Comment by T1mmyK0** [comment]
@stefanobreja I think you need to what for the next version. There is no new version for every small change that was made :)

**Comment by paulfthomas** [comment]
@stefanobreja the fix will be part of the 1.10.0 stable, you can try it out in the [1.10.0-alpha04](https://github.com/material-components/material-components-android/releases/tag/1.10.0-alpha04)
